src/pages/Home/Home.js

```diff
@@ -1,24 +1,126 @@
-import React from 'react';
 import { useHistory } from 'react-router-dom';
+import { useState, useRef } from 'react';
 
-function Home({ userToken, saveToken }) {
+function Home({ createList, joinList }) {
   let history = useHistory();
 
-  function handleClick(e) {
-    saveToken(e);
+  const [shareToken, setShareToken] = useState(''); // value for the shareToken field
+  const shareTokenRef = useRef(); // ref for the shareToken field
+  const [shareTokenError, setShareTokenError] = useState(''); // error hint for the shareToken field
+  const [joinListError, setJoinListError] = useState(''); // error for the entire join list form
 
-    if (userToken) {
-      history.push('/list');
-    }
+  const handleTokenChange = (event) => {
+    setJoinListError('');
+    setShareTokenError('');
+    setShareToken(event.target.value);
+  };
+
+  function handleCreateList() {
+    createList()
+      .then((success) => {
+        history.push('/list');
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+
+  function handleJoinList(event) {
+    event.preventDefault();
+
+    // reset messages to ensure repeated error is read again by screen reader
+    setJoinListError('');
+    setShareTokenError('');
+
+    joinList(shareToken)
+      .then((success) => {
+        history.push('/list');
+      })
+      .catch((err) => {
+        if (err.message === 'Invalid token') {
+          setShareTokenError('Token is invalid.');
+          setJoinListError(
+            'Sorry, there was a problem with your token. Please try again or create a new list.',
+          );
+          shareTokenRef.current.focus();
+        } else {
+          setJoinListError(
+            'Sorry, there was a problem connecting to the database. Please try again.',
+          );
+          shareTokenRef.current.focus();
+        }
+      });
```

Comment:
I like the useRef approach here to put the focus on the input field to encourage the user to enter again. 🥇 for usability.